### ISSUE TITLE
[Feature Request] CDI Support

### ISSUE THREAD

**Issue by 60fpshacksrock** [GerbilSoft/rom-properties#411]
Can you add Disc Juggler CDI support?

**Comment by GerbilSoft** [comment]
I started implementing a CdiReader class some years ago, but never really got around to finishing it. I'll take another look at it later.

**Comment by 60fpshacksrock** [comment]
Thank you.

(ImgBurn supports it if you put the pfctoc.dll beside ImgBurn.exe)

**Comment by GerbilSoft** [comment]
It seems that my old code, and attempts to rework it to handle the proper CDI structures, isn't reading something properly. I'm not sure what's wrong with it, so I'll have to put off adding support for CDI for now.

May look into using another library to read CD formats, e.g. libmirage. (libmirage supports CDI, CCD, NRG, and a bunch of other proprietary disc image formats.)

**Comment by 60fpshacksrock** [comment]
No problem and no rush.

Thank you.

**Comment by GerbilSoft** [comment]
As per #107, the [feature/DiscJuggler-CDI-format.attempt-2](https://github.com/GerbilSoft/rom-properties/tree/feature/DiscJuggler-CDI-format.attempt-2) branch now supports CDI image format for Dreamcast games. This isn't implemented using the generic disc image format handler, since that only works with single-track images.

I'll look into manually implementing CDI support in PlayStationDisc next.

EDIT: Actually, was CDI format ever commonly used for PS1 discs? I can't find any useful references for this...

**Comment by Masamune3210** [comment]
No, not really, it was mostly bin/cue. You of course can find a bunch of different formats that people used along side the half of the population that doesn't realize just how horrible ISO is as a format for anything other than pure data, but its mostly bin/cue/sub and .ecm
